samples/mysql-schema/src/main/java/io/quarkiverse/operatorsdk/samples/mysqlschema/dependent/SecretDependentResource.java

```diff
@@ -0,0 +1,76 @@
+import io.javaoperatorsdk.operator.api.reconciler.dependent.Creator;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.Matcher.Result;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.AssociatedSecondaryResourceIdentifier;
+import io.quarkiverse.operatorsdk.samples.mysqlschema.MySQLSchema;
+
+@ApplicationScoped
+public class SecretDependentResource extends KubernetesDependentResource<Secret, MySQLSchema>
+        implements AssociatedSecondaryResourceIdentifier<MySQLSchema>, Creator<Secret, MySQLSchema> {
+
+    public static final String SECRET_FORMAT = "%s-secret";
+    public static final String USERNAME_FORMAT = "%s-user";
+    public static final String MYSQL_SECRET_USERNAME = "mysql.secret.user.name";
+    public static final String MYSQL_SECRET_PASSWORD = "mysql.secret.user.password";
+
+    private static String encode(String value) {
+        return Base64.getEncoder().encodeToString(value.getBytes());
+    }
+
+    @Override
+    protected Secret desired(MySQLSchema schema, Context context) {
+        final var password = RandomStringUtils
+                .randomAlphanumeric(16); // NOSONAR: we don't need cryptographically-strong randomness here
+        final var name = schema.getMetadata().getName();
+        final var secretName = getSecretName(name);
+        final var userName = String.format(USERNAME_FORMAT, name);
+
+        return new SecretBuilder()
+                .withNewMetadata()
+                .withName(secretName)
+                .withNamespace(schema.getMetadata().getNamespace())
+                .endMetadata()
+                .addToData(MYSQL_SECRET_USERNAME, encode(userName))
+                .addToData(MYSQL_SECRET_PASSWORD, encode(password))
+                .build();
+    }
+
+    private String getSecretName(String name) {
+        return String.format(SECRET_FORMAT, name);
+    }
+
+    @Override
+    public Result<Secret> match(Secret actual, MySQLSchema primary, Context context) {
+        final var desiredSecretName = getSecretName(primary.getMetadata().getName());
+        return Result.nonComputed(actual.getMetadata().getName().equals(desiredSecretName));
+    }
+
+    @Override
+    public ResourceID associatedSecondaryID(MySQLSchema primary) {
+        return new ResourceID(
+                String.format(SECRET_FORMAT, primary.getMetadata().getName()),
+                primary.getMetadata().getNamespace());
+    }
+
+    @Override
+    protected Class<Secret> resourceType() {
+        java.lang.reflect.Type type = getClass().getGenericSuperclass();
+        if (type instanceof Class<?>) {
+            type = ((Class<?>) type).getGenericSuperclass();
+        }
+
+        return (Class<Secret>) ((java.lang.reflect.ParameterizedType) type).getActualTypeArguments()[0];
+    }
+}
```

Comment:
This is mandatory because actual implementation of JOSDK `Utils.getFirstTypeArgumentFromExtendedClass` is not aware of proxy class when using Bean.

Reply:
Why aren't you returning `Secret.class` directly? The reflection is only used in the SDK to provide a default (though flawed) implementation. If you're providing an implementation of that method, you might as well return the actual type that we want, which in this case is `Secret.class`.

Reply:
Oh, I was trying to find a generic solution to be included in the SDK. I don't feel very comfortable having quarkus user override the resourceType when it is already set in every generic parameter. Too bad java don't allow to `return R.class`.